crates/wasmcloud-control-interface/src/lib.rs

```diff
@@ -312,6 +319,47 @@ impl Client {
             Err(e) => Err(format!("Did not receive claims from lattice: {}", e).into()),
         }
     }
+
+    /// Returns the receiver end of a channel that subscribes to the lattice control event stream.
+    /// Any [`PublishedEvent`](struct@PublishedEvent)s that are published after this channel is created
+    /// will be added to the receiver channel's buffer, which can be observed or handled if needed.
+    /// See the example for how you could use this receiver to handle events.
+    ///
+    /// # Example
+    /// ```rust
+    /// use wasmcloud_control_interface::Client;
+    /// async {
+    ///   let nc = nats::asynk::connect("0.0.0.0:4222").await.unwrap();
+    ///   let client = Client::new(nc, None, std::time::Duration::from_millis(1000));
+    ///   let receiver = client.events_receiver().await.unwrap();
+    ///   std::thread::spawn(move || loop {
+    ///     if let Ok(evt) = receiver.recv() {
+    ///       println!("Event received: {:?}", evt);
+    ///     } else {
+    ///       // channel is closed
+    ///       break;
+    ///     }
+    ///   });
+    ///   // perform other operations on client
+    ///   client.get_host_inventory("NAEXHW...").await.unwrap();
+    /// };
+    /// ```
+    pub async fn events_receiver(&self) -> Result<Receiver<PublishedEvent>> {
+        let (sender, receiver) = unbounded();
+        let mut sub = self
+            .nc
+            .subscribe(&broker::control_event(&self.nsprefix))
+            .await?;
+        std::thread::spawn(move || loop {
+            if let Some(msg) = block_on(&mut sub.next()) {
+                match deserialize::<PublishedEvent>(&msg.data) {
+                    Ok(evt) => sender.send(evt).unwrap(),
+                    _ => error!("Object received on event stream was not a PublishedEvent"),
+                }
+            }
+        });
+        Ok(receiver)
+    }
```

Comment:
We should probably add a corresponding function that unsubscribes/clears the event receiver. This makes it so the control interface can choose to be concerned with events for a finite amount of time (or until something meaningful happens according to the developer).

We should also include in the documentation that `events_receiver` creates a subscription and will begin buffering events into a channel so that devs can be aware of the resource cost.

Reply:
Maybe we should make this function more of a verb, e.g. `receive_events` , in which case the corresponding stop/halt function could be called `stop_events` or `ignore_events`.

Reply:
Because each call of `events_receiver` creates a single sender/receiver pair, we may be able to just be explicit here that you can call `drop(receiver)` to close the channel 🤔 I'm going to test that that works before I suggest we go that way.

My only concern with implementing `stop_events` would be that we'd either just `drop` the receiver for them, or we'd have to keep track of the thread that's sending the events so that we can terminate it down the line.

Reply:
I think the concern I have is that dropping the receiver doesn't actually remove the subscription from the nats client, unless you drop the nats client as well.

Reply:
:thinking: it looks like `sub` actually does _move_ into the closure, so the sub should clean itself up. Above `return`, you could explicitly call `sub.unsubscribe();` and that should avoid the continued background CPU activity from the event subscription.

Reply:
re: your comment above (below?) yes I think the `sub` should be dropped, and unsubscribed https://docs.rs/nats/0.8.6/src/nats/asynk/subscription.rs.html#64-68. I think explicitly unsubscribing is a good idea to make sure it's not hanging.

Reply:
nevermind, I just confirmed that NATS will do this for you on drop:
```rust
impl Drop for Inner {
    fn drop(&mut self) {
        self.client.unsubscribe(self.sid).ok();
    }
}
```